crates/storage/db-api/src/models/sharded_key.rs

```diff
@@ -57,30 +53,11 @@ where
     }
 }
 
-impl<T> Decode for ShardedKey<T>
-where
-    T: Decode,
-{
-    fn decode<B: AsRef<[u8]>>(value: B) -> Result<Self, DatabaseError> {
-        let value = value.as_ref();
-
-        let tx_num_index = value.len() - 8;
-
-        let highest_tx_number = u64::from_be_bytes(
-            value[tx_num_index..].try_into().map_err(|_| DatabaseError::Decode)?,
-        );
-        let key = T::decode(&value[..tx_num_index])?;
-
+impl<T: Decode> Decode for ShardedKey<T> {
+    fn decode(value: &[u8]) -> Result<Self, DatabaseError> {
+        let (key, highest_tx_number) = value.split_last_chunk().unwrap();
+        let key = T::decode(key)?;
+        let highest_tx_number = u64::from_be_bytes(*highest_tx_number);
```

Comment:
is this a simplification?

this makes the pr harder to review

ptal @joshieDo

Reply:
skill issue

and `value[tx_num_index..].try_into().map_err(|_| DatabaseError::Decode)?` makes my eyes bleed

Reply:
fine, I'll look at it, but would appreciate these things separately

Reply:
okay this is indeed the same thing but nicer, thanks